tests/utils/nitpick.js

```diff
@@ -521,14 +542,26 @@ module.exports.runRecursive = function () {
     console.warn("Starting recursive tests");
     runningRecursive = true;
 
+    Window.stillSnapshotTaken.connect(onStillSnapshotTaken);
+
     currentRecursiveTestCompleted = true;
     Script.setInterval(
         function () {
             if (currentRecursiveTestCompleted) {
                 currentRecursiveTestCompleted = false;
+
+                // Quit if requested
+                if (quitRequested) {
+                    console.warn("Quitting due to request");
+                    Script.stop();
+                }
+
                 if (testCases.length > 0) {
                     currentTestCase = testCases.pop();
-                    runOneTestCase(currentTestCase, testMode);
+
+                    if (!quitRequested) {

```

Comment:
I don't think `quitRequested` could be set in between line 554 and here, so this might not be necessary.

Reply:
That is what I thought.  But - I think that the script does not stop immediately, and the effect is that when the script does stop - some entities have been created.

Reply:
Perhaps it should `return` after `Script.stop()`?

Reply:
Yep - that works